core/src/main/java/org/apache/iceberg/ManifestGroup.java

```diff
@@ -249,9 +249,11 @@ public CloseableIterator<T> iterator() {
                 .caseSensitive(caseSensitive)
                 .select(columns);
 
-            CloseableIterable<ManifestEntry<DataFile>> entries = reader.entries();
+            CloseableIterable<ManifestEntry<DataFile>> entries = null;
```

Comment:
You can omit the `= null` in this case because it is set in all control flow branches below.